inseq/data/attribution.py

```diff
@@ -474,3 +474,23 @@ class GradientFeatureAttributionStepOutput(FeatureAttributionStepOutput):
     """
 
     _sequence_cls: Type["FeatureAttributionSequenceOutput"] = GradientFeatureAttributionSequenceOutput
+
+
+# Perturbation attribution classes
+
+
+@dataclass(eq=False, repr=False)
+class PerturbationFeatureAttributionSequenceOutput(FeatureAttributionSequenceOutput):
+    """Raw output of a single sequence of perturbation feature attribution."""
+
+    def __post_init__(self):
+        super().__post_init__()
+        self._dict_aggregate_fn["source_attributions"]["sequence_aggregate"] = sum_normalize_attributions
```

Comment:
Do perturbation attributions have shape `[attributed_text_length, generated_text_length, hidden_size]` like the ones generated by gradient methods? `sum_normalize_attributions` ensures to cast the 3D tensor above to a 2D tensor for visualization, but I thought that for occlusion this wouldn't be needed.

If indeed it is not the case, then we would not need a specific class for `PerturbationFeatureAttribution` methods and we could simply stick to the base `FeatureAttributionSequenceOutput` and `FeatureAttributionStepOutput` for the moment.

Reply:
Yes, when I left out this aggregation step from the `__post_init__`, I had a 3D tensor that resulted in a shape violation here: https://github.com/inseq-team/inseq/blob/000aac5acf1a3aba6f558c52dfb5488629b404c0/inseq/data/aggregator.py#L169-L172

I assume this will apply to other perturbation methods as well.

Reply:
I agree with you to have the return object to be of `FeatureAttributionSequenceOutput` and `FeatureAttributionStepOutput` for now.